src/ui/Roster.ts

```diff
@@ -359,19 +360,26 @@ export default class Roster {
    private registerPresenceHandler() {
       this.element.find('.jsxc-js-presence-menu li').click(function() {
          let presenceString = <string> $(this).data('presence');
-         let oldPresence = Client.getPresenceController().getTargetPresence();
-         let requestedPresence = Presence[presenceString];
-
-         if (Client.getAccountManager().getAccount()) {
-            Client.getPresenceController().setTargetPresence(requestedPresence);
+         if (presenceString==='extended-status')
+         {
+            showSetExStatusDialog();
          }
-
-         if (oldPresence === Presence.offline && requestedPresence === Presence.online) {
-            let onUserRequestsToGoOnline = Client.getOption('onUserRequestsToGoOnline');
-
-            if (typeof onUserRequestsToGoOnline === 'function') {
-               onUserRequestsToGoOnline();
-            }
+         else
```

Comment:
Returning inside the if clause would help to reduce indention and enhance the diff too.

Reply:
done